### PR TITLE
add where_or for builder method

### PR DESCRIPTION
**Pull request by ermolaev**

### CODE REVIEW

**Comment by SamSaffron** [comment]
Seems ok to me, mixed on the API name but can not think of anything better. 

@davidtaylorhq  thoughts?

File: lib/mini_sql/builder.rb [hunk 1]
```diff
@@ -83,33 +83,41 @@ def to_sql(hash_args = nil)
...
       end
+    end
+
+    @sections.each do |k, v|
+      joined =
+        case k
+        when :select
+          "SELECT #{v.join(" , ")}"
+        when :where, :where2
+          "WHERE #{v.map { |c| "(#{c})" }.join(" AND ")}"
```
Comment (ermolaev) [inline lib/mini_sql/builder.rb:103]:
```ruby
v = ['user_id = 12', 'order_id = 34', "created_at > '2022-01-01'"]

Benchmark.ips do |x|
  x.report("<<") { (+"WHERE ") << v.map { |c| (+"(") << c << ")" }.join(" AND ") }
  x.report("interpolation") { "WHERE #{v.map { |c| "(#{c})"}.join(" AND ")}" }
  x.compare!
end

# Comparison:
#    interpolation:  1158228.3 i/s
#               <<:  1064932.3 i/s - 1.09x  (± 0.00) slower
```

File: test/mini_sql/builder_tests.rb [hunk 2]
```diff
@@ -161,6 +161,26 @@ def test_query_decorator
...
+
+  def test_where_or_and
+    builder = @connection.build("SELECT 1 as one /*where*/")
+    builder
+      .where_or("1 = ?", 2)
+      .where_or("1 = ?", 1)
+      .where("3 = ?", 3)
+      .where("4 = ?", 4)
+
+    assert_equal(builder.to_sql, "SELECT 1 as one WHERE (3 = 3) AND (4 = 4) AND ((1 = 2) OR (1 = 1))")
```
Comment (ermolaev) [inline test/mini_sql/builder_tests.rb:181]: now we can build more difficult conditions

**Comment by ermolaev** [comment]
@SamSaffron, do you have any additions on this PR? Can it be merged?

**Comment by ermolaev** [comment]
@SamSaffron, do you have any additions on this PR? Can it be merged?

**Review by SamSaffron — APPROVED** [review]

**Comment by SamSaffron** [comment]
Looks good to me!

**Pull request merged**